source/extensions/filters/network/redis_proxy/conn_pool_impl.cc

```diff
@@ -48,7 +48,8 @@ InstanceImpl::makeRequestToHost(const std::string& host_address,
 
 InstanceImpl::ThreadLocalPool::ThreadLocalPool(InstanceImpl& parent, Event::Dispatcher& dispatcher,
                                                std::string cluster_name)
-    : parent_(parent), dispatcher_(dispatcher), cluster_name_(std::move(cluster_name)) {
+    : parent_(parent), dispatcher_(dispatcher), cluster_name_(std::move(cluster_name)),
+      drain_timer_(dispatcher.createTimer([this]() -> void { drainClients(); })) {
```

Comment:
Are the drain_timer_ executing drainClients() on the main thread or the worker thread?  If it's on the worker thread then we might have contention with the make_requests calls.

Reply:
drainClients() is specific to ThreadLocalPool and executes on the worker thread's dispatcher -- same as makeRequest..(). The timer is only enabled when clients are put aside to drain, and it only stays enabled as long as there are draining clients with outstanding requests. So, it is a bit of extra activity, but it shouldn't be a problem given the relative infrequency of this activity.

Reply:
Sounds good.